ffi/athcon/bindings/rust/athcon-vm/src/encode_tx.rs

```diff
@@ -0,0 +1,137 @@
+//! Implements athcon_encode_tx
+use athcon_sys as ffi;
+use athena_interface::transaction::{Encode, Transaction};
+use athena_vm_sdk::{Pubkey, SendArguments};
+
+/// Encode Athena Transaction into bytes.
+///
+/// # Arguments
+/// - `principal` - The transaction's principal account.
+/// - `template` - Optional transaction's template account.
+/// - (`method`, `method_size`) - Optional method to call
+/// - `nonce` - The transaction's nonce
+/// - (`args`, `args_size`) - Optional transaction's serialized arguments.
+///
+/// # Safety
+/// - the caller is responsible for freeing the returned vector
+///   via the `athcon_free_vector` function.
+#[no_mangle]
+unsafe extern "C" fn athcon_encode_tx(
```

Comment:
I think we can do away with this method entirely and just keep the two encode txs below, one for spawn, one for spend. They need to include method as well as args.